hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileUtil.java

```diff
@@ -715,6 +746,9 @@ public static void unZip(File inFile, File unzipDir) throws IOException {
               }
             } finally {
               out.close();
+              if (entry.getPlatform() == ZipArchiveEntry.PLATFORM_UNIX) {
```

Comment:
this is going to execute even if writing to out failed. not sure that is ideal.
better to put it on a line after the try/finally block. so is only called if the write succeeded.

Reply:
This is a good suggestion and I will revise it.